src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Services/SyntaxFacts/CSharpSyntaxKinds.cs

```diff
@@ -39,6 +39,8 @@ protected CSharpSyntaxKinds()
         public int RegionDirectiveTrivia => (int)SyntaxKind.RegionDirectiveTrivia;
         public int EndRegionDirectiveTrivia => (int)SyntaxKind.EndRegionDirectiveTrivia;
         public int? ShebangDirectiveTrivia => (int)SyntaxKind.ShebangDirectiveTrivia;
+        public int DefineDirectiveTrivia => (int)SyntaxKind.DefineDirectiveTrivia;
+        public int? UndefDirectiveTrivia => (int)SyntaxKind.UndefDirectiveTrivia;
```

Comment:
why is undef nullable?

Reply:
VB does not have an undef directive, you only define pp symbols through `#Const`, which is the `DefineDirectiveTrivia` implementation